lib/theme/widgets/hs_date_picker.dart

```diff
@@ -1,34 +1,40 @@
+import 'package:clock/clock.dart';
 import 'package:flutter/material.dart';
+import 'package:hatspace/singleton/hs_singleton.dart';
 import 'package:hatspace/theme/hs_date_picker_theme.dart';
 import 'package:table_calendar/table_calendar.dart';
 
 class HsDatePicker extends StatelessWidget {
-  final ValueNotifier<DateTime> selectedDate;
-  final ValueChanged saveSelectDate;
-  const HsDatePicker(
-      {required this.selectedDate, required this.saveSelectDate, super.key});
+  final ValueChanged<DateTime> saveSelectDate;
+
+  late final ValueNotifier<DateTime> _selectedDateNotifier;
```

Comment:
- dispose `_selectedDateNotifier`: will fix
- the use of ValueNotifier to is to update this UI itself, without the need to hook the state from outside of the view. This will help to keep this implementation isolated from other components.